src/utils/external-extensions.ts

```diff
@@ -44,3 +45,30 @@ export const getArgumentFromExternalExtensionOption = (externalExtensionOption:
 
   return `${owner}/${project}${branch ? `:${branch}` : ""}`;
 };
+
+// Gets the solidity framework directories from the external extension repository
+export const getSolidityFrameworkDirsFromExternalExtension = async ({
+  repositoryURL,
+  branch,
+}: {
+  repositoryURL: string;
+  branch?: string;
+}) => {
+  const splitUrl = repositoryURL.split("/");
+  const ownerName = splitUrl[splitUrl.length - 2];
+  const repoName = splitUrl[splitUrl.length - 1];
+  const githubApiUrl = `https://api.github.com/repos/${ownerName}/${repoName}/contents/extension/packages${branch ? `?ref=${branch}` : ""}`;
+  const res = await fetch(githubApiUrl);
```

Comment:
Since we are making unauthenticated request, github has limit on it which is 

> The primary rate limit for unauthenticated requests is 60 requests per hour.
~ [api docs](https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#primary-rate-limit-for-unauthenticated-users)

I think it should be almost good enough?

Reply:
We are making a a couple of requests (+ clone later) so it should be fine.

If in the future we see some issue, maybe we can just clone the repo at the beginning and read locally.